### DISCUSSION TITLE
Adding secp signatures in addition to ed25519 signatures

### DISCUSSION THREAD

**Discussion by bowenwang1996** [near/NEPs#140] [Ideas]
@SkidanovAlex mentioned that the bridge could benefit a lot from having secp signatures in block headers in addition to ed25519 signatures because it is cheaper to verify secp signatures on ethereum side and we don't have to rely on challenges for the bridge to work. In addition, having secp signatures in blocks allows us to build a bitcoin bridge in the future. However, the downside is quite obvious: the change is very invasive since it not only changes the block structure, but we also need to introduce a new action for staking with secp keys (or modify the existing staking action, which is less desirable).

**Comment by ilblackdragon** [comment]
I'm not sure that we need to add staking action with separate secp key.

Because the ed25519 is still the main key, if message is signed with ed25519 key - you can trust the secp key.
Also secp key can be inferred from secp signature via ecrecovery process.

Are there ways we can allow validator to add a message ahead of their epoch with secp signature of key they are going to use?
We can also require this to make sure validators that staked are online before their epoch, otherwise they still not get slot? 
E.g. solving the problem that I can stake but if my node is offline (still syncing or whatever) - we will still allocate that node to be validator.

PS. we really need to figure out where we are discussing what. E.g. this is a protocol change, and seems like should be discussed in NEPs?

**Reply by bowenwang1996** [reply]
I don't quite understand. There needs to be a way for validators to register some secp key. Are you suggesting deriving the secp key from their ed25519 key?

**Comment by bowenwang1996** [comment]
Consolidated discussions in https://gov.near.org/t/add-secp-validator-key-to-enable-pessimistic-bridge/401
